src/Microsoft.AspNetCore.Mvc.Core/Internal/RedirectToActionResultExecutor.cs

```diff
@@ -46,7 +47,24 @@ public void Execute(ActionContext context, RedirectToActionResult result)
             }
 
             _logger.RedirectToActionResultExecuting(destinationUrl);
-            context.HttpContext.Response.Redirect(destinationUrl, result.Permanent);
+            if (result.PreserveMethod == false)
+            {
+                context.HttpContext.Response.Redirect(destinationUrl, result.Permanent);
+            }
+
+            else
+            {
```

Comment:
most of this code is due to all of the conditionals. There are only two lines you need for any redirect:
```
response.StatusCode = StatusCodes.Status308PermanentRedirect; // Code as needed
response.Headers[HeaderNames.Location] = destinationUrl;
```